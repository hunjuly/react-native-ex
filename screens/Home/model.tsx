import React from 'react'
import { RootTabScreenProps } from '@/navigation/types'

export type Props = RootTabScreenProps<'Home'>

export function useModel(P: Props) {
    const { navigation } = P

    const [modalVisible, setModalVisible] = React.useState(false)

    const showPopup = () => {
        setModalVisible(true)
    }

    const hidePopup = () => {
        setModalVisible(false)
    }

    return {
        modalVisible,
        showPopup,
        hidePopup
    }
}
