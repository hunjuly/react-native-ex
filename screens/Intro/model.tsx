import React from 'react'
import { RootStackScreenProps } from '@/navigation/types'

export type Props = RootStackScreenProps<'Intro'>

export function useModel(P: Props) {
    const { navigation } = P

    const [isFocused, setFocused] = React.useState(navigation.isFocused())

    const showLookAround = () => alert('look around')
    const onLogin = () => alert('login')

    React.useEffect(() => {
        const unsubscribe = navigation.addListener('state', () => {
            setFocused(navigation.isFocused())
        })

        return unsubscribe
    }, [navigation])

    return {
        isFocused,
        showLookAround,
        onLogin
    }
}
