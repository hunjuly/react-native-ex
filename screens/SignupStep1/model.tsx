import React from 'react'
import { TextInput } from 'react-native'
import { isValidEmail } from '@/common'
import { RootStackScreenProps } from '@/navigation/types'

export type Props = RootStackScreenProps<'SignupStep1'>

export function useModel(P: Props) {
    const { navigation } = P

    const [email, setEmail] = React.useState(__DEV__ ? 'user@test.com' : '')
    const [marketingAgreement, setMarketingAgreement] = React.useState(false)

    // states
    const emailRef = React.createRef<TextInput>()

    const [policyAgreement, setPolicyAgreement] = React.useState(false)
    const [emailDenied, setEmailDenied] = React.useState(false)
    const [emailFocused, setEmailFocused] = React.useState(false)

    const emailAlerted = !isValidEmail(email) && !emailFocused
    const canContinue = policyAgreement && isValidEmail(email) && !emailDenied

    // callbacks
    const doContinue = async () => {
        try {
            navigation.navigate('SignupStep2', { email, marketingAgreement })
        } catch (error) {
            alert(error)
        }
    }

    const showPolicy = () => {
        alert('showPolicy')
    }

    const onEmailChanged = (text: string) => {
        setEmailDenied(false)
        setEmail(text)
    }

    // Login -> Signup의 경우도 있어서 navigation.goBack()을 하지 않음
    const goBack = () => {
        navigation.navigate('Intro')
    }

    React.useEffect(() => {
        emailRef.current?.focus()
    }, [])

    return {
        email,
        setMarketingAgreement,
        marketingAgreement,
        emailAlerted,
        canContinue,
        doContinue,
        showPolicy,
        onEmailChanged,
        setEmailFocused,
        setPolicyAgreement,
        policyAgreement,
        emailRef,
        goBack
    }
}
