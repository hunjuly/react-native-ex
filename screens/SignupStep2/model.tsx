import React from 'react'
import { useRequestSignup, useVerifySignup } from '@/apis'
import { RequestContext } from '@/hooks'
import { RootStackScreenProps } from '@/navigation/types'

export type Props = RootStackScreenProps<'SignupStep2'>

export function useModel(P: Props) {
    const { navigation, route } = P
    const { email, authId, marketingAgreement } = route.params

    const request = React.useContext(RequestContext)
    const verifySignup = useVerifySignup(request)
    const requestSignup = useRequestSignup(request)

    const checkAuthCode = async (code: string) => verifySignup(email, code, authId)

    const sendAuthCodeEmail = async () => requestSignup(email)

    const onSuccess = async () => navigation.navigate('SignupStep3', { email, authId, marketingAgreement })

    const goBack = () => navigation.goBack()

    return {
        email,
        goBack,
        sendAuthCodeEmail,
        checkAuthCode,
        onSuccess
    }
}
