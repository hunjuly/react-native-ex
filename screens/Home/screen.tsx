import React from 'react'
import { SafeScrollContainer } from '@/components'
import { Button } from '@/theme'
import { ExamplePopup } from './ExamplePopup'
import { Props, useModel } from './model'
import { useStyles, useTexts } from './resource'

export function Home(P: Props) {
    const M = useModel(P)
    const S = useStyles()
    const T = useTexts()

    const [modalVisible, setModalVisible] = React.useState(true)

    return (
        <SafeScrollContainer style={S.container}>
            <ExamplePopup visible={modalVisible} setVisible={setModalVisible} />
            <Button.TextL
                title="FadeModal popup"
                onPress={() => {
                    setModalVisible(!modalVisible)
                }}
            />
            <Button.TextL title="Modal popup" onPress={() => {}} />
        </SafeScrollContainer>
    )
}
