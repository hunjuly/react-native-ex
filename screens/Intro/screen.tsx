import React from 'react'
import { View } from 'react-native'
import { useWindowDimensions } from 'react-native'
import { StatusBar } from 'expo-status-bar'
import { PageIndicator, PageRange, PageScrollView, SafeAreaBottom } from '@/components'
import { FirstPage } from './FirstPage'
import { LookAround } from './LookAround'
import { OtherPage } from './OtherPage'
import { Props, useModel } from './model'
import { useStyles, useTexts } from './resource'

const image1 = require('@/assets/images/intro/page1.png')
const image2 = require('@/assets/images/intro/page2.png')
const image3 = require('@/assets/images/intro/page3.png')

export function Intro(P: Props) {
    const S = useStyles()
    const T = useTexts()
    const M = useModel(P)

    // screenWidth가 아니라 PageScrollView의 onLayout을 받아서 width를 설정해야 한다.
    const screenWidth = useWindowDimensions().width

    const [range, setPageRange] = React.useState<PageRange>({ total: 0, current: 0 })

    const statusBarStyle = range.current === 0 ? 'light' : 'dark'

    return (
        <View style={S.main.container}>
            <StatusBar style={statusBarStyle} />
            <PageScrollView
                style={S.main.scrollView}
                pageWidth={screenWidth}
                range={range}
                setPageRange={setPageRange}
                isFocused={M.isFocused}
            >
                <FirstPage />
                <OtherPage text={T.page1} image={image1} />
                <OtherPage text={T.page2} image={image2} />
                <OtherPage text={T.page3} image={image3} />
            </PageScrollView>
            <PageIndicator style={{ marginBottom: 16 }} range={range} />
            <LookAround range={range} onPress={M.showLookAround} />
            <SafeAreaBottom />
        </View>
    )
}
