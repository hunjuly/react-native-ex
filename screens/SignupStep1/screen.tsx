import React from 'react'
import { View } from 'react-native'
import { StatusBar } from 'expo-status-bar'
import {
    Button,
    CheckBox,
    EmailField,
    KeyboardAnimateView,
    KeyboardAnimated,
    NavigationTitleBar,
    SafeAreaView,
    Text
} from '@/components'
import { useStepStyles } from '@/components/styles'
import { RootStackScreenProps } from '@/navigation/types'
import { Props, useModel } from './model'
import { useStyles, useTexts } from './resource'

export function SignupStep1(P: Props) {
    const M = useModel(P)
    const S = useStyles()
    const T = useTexts()
    const Step = useStepStyles()

    return (
        <View style={[Step.container]}>
            <StatusBar style={'dark'} />
            <SafeAreaView>
                <View style={Step.headerBox}>
                    <NavigationTitleBar onBackButton={M.goBack} title={T.title} />
                    <View style={Step.stepBox}>
                        <Text.Subtext style={Step.stepText} value={T.stepOf} />
                        <View style={Step.stepBarBox}>
                            <View style={Step.activeBar} />
                            <View style={Step.deactiveBar} />
                            <View style={Step.deactiveBar} />
                            <View style={Step.deactiveBar} />
                        </View>
                    </View>
                </View>

                <KeyboardAnimateView navigation={P.navigation} style={S.keyboardAnimateView}>
                    <Text.H1 style={S.title} value={T.subtitle} />
                    <Text.Body1 style={S.desc} value={T.desc} />
                    <EmailField
                        style={S.emailField}
                        forwardedRef={M.emailRef}
                        onChangeFocus={M.setEmailFocused}
                        currentText={M.email}
                        placeholder={T.placeholder}
                        invalid={M.emailAlerted}
                        enableCorrectMark={true}
                        onChange={M.onEmailChanged}
                    />
                    <Text.Caption style={S.invalidEmail} value={M.emailAlerted && T.invalidEmail} />
                    <CheckBox
                        onChange={M.setPolicyAgreement}
                        checked={M.policyAgreement}
                        text={T.policyAgreement(M.showPolicy)}
                        textStyle={S.agreementText}
                    />
                    <CheckBox
                        testID="marketingAgreement"
                        style={S.marketingAgreement}
                        onChange={M.setMarketingAgreement}
                        checked={M.marketingAgreement}
                        text={T.marketingAgreement}
                        textStyle={S.agreementText}
                    />
                    <KeyboardAnimated>
                        <Button.PrimaryL
                            title={T.continue}
                            style={S.continue}
                            onPress={M.doContinue}
                            disabled={!M.canContinue}
                        />
                    </KeyboardAnimated>
                </KeyboardAnimateView>
            </SafeAreaView>
        </View>
    )
}
