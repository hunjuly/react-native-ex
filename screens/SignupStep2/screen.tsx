import React from 'react'
import { View } from 'react-native'
import { StatusBar } from 'expo-status-bar'
import { NavigationTitleBar, SafeAreaView } from '@/components'
import { Text, useStepStyles } from '@/theme'
import { VerifyEmailCode } from '@/views'
import { Props, useModel } from './model'
import { useTexts } from './resource'

export function SignupStep2(P: Props) {
    const M = useModel(P)
    const T = useTexts()
    const Step = useStepStyles()

    return (
        <View style={[Step.container]}>
            <StatusBar style={'dark'} />
            <SafeAreaView>
                <View style={Step.headerBox}>
                    <NavigationTitleBar onBackButton={M.goBack} title={T.title} />
                    <View style={Step.stepBox}>
                        <Text.Body1 style={Step.stepText} value={T.stepOf} />
                        <View style={Step.stepBarBox}>
                            <View style={Step.activeBar} />
                            <View style={Step.activeBar} />
                            <View style={Step.deactiveBar} />
                            <View style={Step.deactiveBar} />
                        </View>
                    </View>
                </View>
                <VerifyEmailCode
                    email={M.email}
                    sendAuthCodeEmail={M.sendAuthCodeEmail}
                    checkAuthCode={M.checkAuthCode}
                    onSuccess={M.onSuccess}
                    navigation={P.navigation}
                />
            </SafeAreaView>
        </View>
    )
}
