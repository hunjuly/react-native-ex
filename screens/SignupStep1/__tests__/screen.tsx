import { render, screen } from '@testing-library/react-native'
import * as React from 'react'
import { Props } from '../model'
import { SignupStep1 } from '../screen'

let mockModel = {}

jest.mock('../model', () => ({
    useModel: () => mockModel
}))

describe('GoogleOtp screen', () => {
    function renderScreen(values: any) {
        mockModel = {
            isError: false,
            ...values
        }

        const P = {
            navigation: { addListener: () => {} }
        } as unknown as Props

        render(<SignupStep1 {...P} />)
    }

    test('default states', async () => {
        renderScreen({})

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test('all true states ', async () => {
        const message = 'test message'

        renderScreen({
            isError: true
        })

        const displayed = existText(message)
        expect(displayed).toBeTruthy()

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
