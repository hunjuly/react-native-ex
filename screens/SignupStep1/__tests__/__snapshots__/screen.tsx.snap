// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleOtp screen all true states  1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 44,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#020725",
                },
                Object {
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 28,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32.8,
                },
                Object {
                  "color": "#000000",
                  "flex": 1,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign up
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "marginHorizontal": 22,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#020725",
              },
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 0.5,
                "lineHeight": 20,
              },
              Object {
                "color": "#04285A",
              },
            ]
          }
        >
          STEP 1 OF 4
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 4,
              "marginHorizontal": 0,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04285A",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        keyboardDismissMode="interactive"
        keyboardShouldPersistTaps="handled"
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        style={
          Object {
            "marginHorizontal": 22,
          }
        }
      >
        <View>
          <View
            onLayout={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#020725",
                  },
                  Object {
                    "fontFamily": "Roboto-Bold",
                    "fontSize": 32,
                    "fontWeight": "700",
                    "letterSpacing": 0.25,
                    "lineHeight": 37.5,
                  },
                  Object {
                    "color": "#000000",
                    "marginTop": 32,
                  },
                ]
              }
            >
              What is your email address?
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#020725",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#59554C",
                    "marginTop": 6,
                  },
                ]
              }
            >
              You’ll use this email to log in.
            </Text>
            <View
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "marginTop": 17,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                defaultValue=""
                keyboardType="email-address"
                onEndEditing={[Function]}
                onFocus={[Function]}
                placeholder="example_email@address.com"
                placeholderTextColor="#ACACAC"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F3F3F4",
                      "borderBottomColor": "#04285A",
                      "borderBottomWidth": 2,
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                      "color": "#59554C",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "height": 48,
                      "letterSpacing": 0.5,
                      "lineHeight": 20,
                      "paddingHorizontal": 14,
                    },
                  ]
                }
              />
              <Image
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "position": "absolute",
                    "right": 8,
                  }
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#CA433C",
                    "textAlign": "right",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Image
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": 14,
                    "marginTop": 0,
                    "overflow": "hidden",
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#59554C",
                      },
                    ]
                  }
                >
                  By submitting your email, you confirm you’ve read the 
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#020725",
                        },
                        Object {
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.5,
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#063576",
                        },
                      ]
                    }
                  >
                    GROW Policy
                  </Text>
                  .
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  Object {
                    "marginTop": 17,
                  },
                ]
              }
            >
              <Image
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": 14,
                    "marginTop": 0,
                    "overflow": "hidden",
                  }
                }
                testID="marketingAgreement"
              />
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#59554C",
                      },
                    ]
                  }
                >
                  I would like to receive marketing information (optional).
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "marginTop": 0,
              }
            }
          >
            <View
              onLayout={[Function]}
            >
              <View>
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 10,
                        "borderStyle": "solid",
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#EDEDED",
                        "textColor": "#ACACAC",
                      },
                      Object {
                        "alignSelf": "stretch",
                        "height": 50,
                        "marginBottom": 26,
                        "marginHorizontal": 2,
                        "marginTop": 13,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "letterSpacing": 0.18,
                          "lineHeight": 30,
                        },
                        Object {
                          "color": "#ACACAC",
                        },
                      ]
                    }
                  >
                    Continue
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`GoogleOtp screen default states 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 44,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#020725",
                },
                Object {
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 28,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 32.8,
                },
                Object {
                  "color": "#000000",
                  "flex": 1,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign up
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": "100%",
                "position": "absolute",
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 30,
            "marginHorizontal": 22,
            "marginTop": 16,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#020725",
              },
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 0.5,
                "lineHeight": 20,
              },
              Object {
                "color": "#04285A",
              },
            ]
          }
        >
          STEP 1 OF 4
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 4,
              "marginHorizontal": 0,
              "marginTop": 8,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#04285A",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#EAEEF3",
                "flex": 1,
                "height": 4,
                "marginHorizontal": 2,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        keyboardDismissMode="interactive"
        keyboardShouldPersistTaps="handled"
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        style={
          Object {
            "marginHorizontal": 22,
          }
        }
      >
        <View>
          <View
            onLayout={[Function]}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#020725",
                  },
                  Object {
                    "fontFamily": "Roboto-Bold",
                    "fontSize": 32,
                    "fontWeight": "700",
                    "letterSpacing": 0.25,
                    "lineHeight": 37.5,
                  },
                  Object {
                    "color": "#000000",
                    "marginTop": 32,
                  },
                ]
              }
            >
              What is your email address?
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#020725",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#59554C",
                    "marginTop": 6,
                  },
                ]
              }
            >
              You’ll use this email to log in.
            </Text>
            <View
              style={
                Array [
                  Object {
                    "justifyContent": "center",
                  },
                  Object {
                    "marginTop": 17,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                defaultValue=""
                keyboardType="email-address"
                onEndEditing={[Function]}
                onFocus={[Function]}
                placeholder="example_email@address.com"
                placeholderTextColor="#ACACAC"
                style={
                  Array [
                    Object {
                      "backgroundColor": "#F3F3F4",
                      "borderBottomColor": "#04285A",
                      "borderBottomWidth": 2,
                      "borderTopLeftRadius": 4,
                      "borderTopRightRadius": 4,
                      "color": "#59554C",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "height": 48,
                      "letterSpacing": 0.5,
                      "lineHeight": 20,
                      "paddingHorizontal": 14,
                    },
                  ]
                }
              />
              <Image
                accessible={true}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "position": "absolute",
                    "right": 8,
                  }
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.5,
                    "lineHeight": 20,
                  },
                  Object {
                    "color": "#CA433C",
                    "textAlign": "right",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Image
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": 14,
                    "marginTop": 0,
                    "overflow": "hidden",
                  }
                }
              />
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#59554C",
                      },
                    ]
                  }
                >
                  By submitting your email, you confirm you’ve read the 
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#020725",
                        },
                        Object {
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.5,
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#063576",
                        },
                      ]
                    }
                  >
                    GROW Policy
                  </Text>
                  .
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  Object {
                    "marginTop": 17,
                  },
                ]
              }
            >
              <Image
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                source={1}
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": 14,
                    "marginTop": 0,
                    "overflow": "hidden",
                  }
                }
                testID="marketingAgreement"
              />
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 15,
                        "fontWeight": "400",
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#59554C",
                      },
                    ]
                  }
                >
                  I would like to receive marketing information (optional).
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              Object {
                "marginTop": 0,
              }
            }
          >
            <View
              onLayout={[Function]}
            >
              <View>
                <View
                  accessibilityState={
                    Object {
                      "disabled": true,
                    }
                  }
                  accessible={true}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 10,
                        "borderStyle": "solid",
                        "justifyContent": "center",
                      },
                      Object {
                        "backgroundColor": "#EDEDED",
                        "textColor": "#ACACAC",
                      },
                      Object {
                        "alignSelf": "stretch",
                        "height": 50,
                        "marginBottom": 26,
                        "marginHorizontal": 2,
                        "marginTop": 13,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "letterSpacing": 0.18,
                          "lineHeight": 30,
                        },
                        Object {
                          "color": "#ACACAC",
                        },
                      ]
                    }
                  >
                    Continue
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</View>
`;
